kotlin/legacy/legacy-workflow-rx2/src/main/java/com/squareup/workflow/legacy/rx2/EventChannel.kt

```diff
@@ -123,7 +123,7 @@ fun <E : Any> ReceiveChannel<E>.asEventChannel() = object : EventChannel<E> {
             .also { selectionJob.cancel() }
       } catch (cancellation: CancellationException) {
         val cause = cancellation.cause
-        if (cause == null) {
+        if (cause == null || cause is CancellationException) {
```

Comment:
Normal cancellations now have a self-referential cause, so we can't just check for null.